packages/content/test/index.spec.ts

```diff
@@ -0,0 +1,199 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+import * as assert from 'assert';
+import { ContentStoreService, NavigationService, UrlTransformationService } from '@craftercms/content';
+import { crafterConf } from '@craftercms/classes';
+import 'url-search-params-polyfill';
+
+import mock from 'xhr-mock';
+import MockRequest from "xhr-mock/lib/MockRequest";
+import MockResponse from "xhr-mock/lib/MockResponse";
+
+import { 
+  item,
+  descriptor,
+  children,
+  tree,
+  navTree,
+  navBreadcrumb,
+  renderUrl,
+  storeUrl
+} from './mock-responses';
+
+crafterConf.configure({
+  site: 'editorial'
+})
+
+describe('Engine Client', () => {
+
+  // replace the real XHR object with the mock XHR object before each test
+  beforeEach(() => mock.setup());
+
+  // put the real XHR object back and clear the mocks after each test
+  afterEach(() => mock.teardown());
+
+  describe('Content Store Service', () => {
+
+    describe('getItem', () => {
+      it('should return the index item', done => {
+
+        mock.get("http://localhost:8080/api/1/site/content_store/item.json?url=%2Fsite%2Fwebsite%2Findex.xml&crafterSite=editorial", 
+        (req: MockRequest, res: MockResponse) => {
+          res.body(JSON.stringify(item));
+          return res;
+        });
+
+        ContentStoreService.getItem("/site/website/index.xml", crafterConf.getConfig())
+          .subscribe((item) => {
+            assert.equal(item.descriptorDom.page.objectId, '8d7f21fa-5e09-00aa-8340-853b7db302da');
+            done();
+          })
+      });
+    });
+
+    describe('getDescriptor', () => {
+      it('should return the index descriptor', done => {
+
+        mock.get("http://localhost:8080/api/1/site/content_store/descriptor.json?url=%2Fsite%2Fwebsite%2Findex.xml&crafterSite=editorial", 
+        (req: MockRequest, res: MockResponse) => {
+          res.body(JSON.stringify(descriptor));
+          return res;
+        });
+
+        ContentStoreService.getDescriptor("/site/website/index.xml", crafterConf.getConfig())
+          .subscribe((descriptor) => {
+            assert.equal(descriptor.page.objectId, '8d7f21fa-5e09-00aa-8340-853b7db302da');
+            done();
+          })
+      });
+    });
+
+    describe('getChildren', () => {
+      it('should return the child pages', done => {
+
+        mock.get("http://localhost:8080/api/1/site/content_store/children.json?url=%2Fsite%2Fwebsite%2F&crafterSite=editorial", 
+        (req: MockRequest, res: MockResponse) => {
+          res.body(JSON.stringify(children));
+          return res;
+        });
+
+        ContentStoreService.getChildren("/site/website/", crafterConf.getConfig())
+          .subscribe((children) => {
+            assert.equal(children.length, 9, 'index should have 9 child pages');
```

Comment:
The `9` here (same as similar tests) is also a case where we can read the value from the mocked object